File: src/main/java/no/nav/bidrag/beregn/bidragsevne/BidragsevneCore.java
package no.nav.bidrag.beregn.bidragsevne;

import no.nav.bidrag.beregn.bidragsevne.dto.BeregnBidragsevneGrunnlagCore;
import no.nav.bidrag.beregn.bidragsevne.dto.BeregnetBidragsevneResultatCore;
import no.nav.bidrag.beregn.bidragsevne.periode.BidragsevnePeriode;

public interface BidragsevneCore {

  BeregnetBidragsevneResultatCore beregnBidragsevne(
      BeregnBidragsevneGrunnlagCore beregnBidragsevneGrunnlagCore);

  static BidragsevneCore getInstance() { return new BidragsevneCoreImpl(
      BidragsevnePeriode.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/bidragsevne/BidragsevneCore.kt
package no.nav.bidrag.beregn.bidragsevne

import no.nav.bidrag.beregn.bidragsevne.dto.BeregnBidragsevneGrunnlagCore
import no.nav.bidrag.beregn.bidragsevne.dto.BeregnetBidragsevneResultatCore
import no.nav.bidrag.beregn.bidragsevne.periode.BidragsevnePeriode

fun interface BidragsevneCore {
    fun beregnBidragsevne(grunnlag: BeregnBidragsevneGrunnlagCore): BeregnetBidragsevneResultatCore

    companion object {
        fun getInstance(): BidragsevneCore = BidragsevneCoreImpl(BidragsevnePeriode.getInstance())
    }
}
